### DISCUSSION TITLE
Getting Started

### DISCUSSION THREAD

**Discussion by Abstaina44** [Cyfrin/foundry-full-course-cu#1760] [General]
This question has been bothering me and I need to ask now, it is a must to have JavaScript Knowledge before going into solidity and blockchains( Smart Contracts)..
I am not new but I have a little knowledge in web development ( front- end).

**Comment by EngrPips** [comment]
It is not a must, but prior programming knowledge is definitely a plus. The courses on Cyfrin Updraft cover everything you need to be a smart contract developer/auditor. If you are new, start with `Blockchain Basics` and progress through the courses on the platform.

**Comment by urosognjenovic** [comment]
JavaScript is necessary if you're using Hardhat which is JS based and if you want to learn how to interact with the front end. 

So you're good to go without it.
